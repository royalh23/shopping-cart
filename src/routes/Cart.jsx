import styles from '../styles/Cart.module.css';
import { Link, useOutletContext } from 'react-router-dom';

export default function Cart() {
  const [cartItems, setCartItems] = useOutletContext();

  return (
    <main className={styles.main}>
      {cartItems.length === 0 ? (
        <>
          <p className={styles.para}>Your shopping cart is empty.</p>
          <Link to="/shopping-items" className={styles.link}>
            Start shopping
          </Link>
        </>
      ) : (
        <div></div>
      )}
    </main>
  );
}
