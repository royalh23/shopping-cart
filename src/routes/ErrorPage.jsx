import { Link } from 'react-router-dom';
import styles from '../styles/ErrorPage.module.css';

export default function ErrorPage() {
  return (
    <div className={styles.errorPage} data-testid="error-page">
      <h1>Oops, page not found</h1>
      <p>
        Click <Link to="/">here</Link> to get back to homepage.
      </p>
    </div>
  );
}
