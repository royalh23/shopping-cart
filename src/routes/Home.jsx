import { Link } from 'react-router-dom';
import styles from '../styles/Home.module.css';
import maleShopper from '../assets/male-shopper.jpeg';

export default function Home() {
  return (
    <main className={styles.main}>
      <div>
        <h1 className={styles.heading}>
          A store where you can find everything!
        </h1>
        <p className={styles.para}>
          Anything you need, at your fingertips. Take a look!
        </p>
        <Link to="shopping-items" className={styles.link}>
          Start shopping
        </Link>
      </div>
      <div className={styles.mainImg}>
        <img src={maleShopper} alt="Male shopper" />
      </div>
    </main>
  );
}
