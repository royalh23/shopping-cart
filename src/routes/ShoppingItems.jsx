import ShoppingItem from '../components/ShoppingItem';
import useItems from '../hooks/use-items';
import styles from '../styles/ShoppingItems.module.css';

export default function ShoppingItems() {
  const { items, loading, error } = useItems();

  return (
    <main className={styles.main}>
      {loading ? (
        <p className={styles.loading}>Loading...</p>
      ) : error ? (
        <p className={styles.error}>{error}</p>
      ) : (
        <>
          <h1 className={styles.heading}>Items</h1>
          <div className={styles.items}>
            {items.map((item) => (
              <ShoppingItem
                key={item.id}
                title={item.title}
                price={item.price}
                imageURL={item.imageURL}
                id={item.id}
              />
            ))}
          </div>
        </>
      )}
    </main>
  );
}
