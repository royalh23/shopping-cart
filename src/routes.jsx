import App from './App';
import Home from './routes/Home';
import ShoppingItems from './routes/ShoppingItems';
import Cart from './routes/Cart';
import ErrorPage from './routes/ErrorPage';

const routes = [
  {
    path: '/',
    element: <App />,
    errorElement: <ErrorPage />,
    children: [
      { index: true, element: <Home /> },
      { path: 'shopping-items', element: <ShoppingItems /> },
      { path: 'cart', element: <Cart /> },
    ],
  },
];

export default routes;
