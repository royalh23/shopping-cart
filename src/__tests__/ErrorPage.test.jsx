import { createMemoryRouter } from 'react-router-dom';
import routes from '../routes';
import { RouterProvider } from 'react-router';
import { render, screen } from '@testing-library/react';
import { vi } from 'vitest';
import userEvent from '@testing-library/user-event';

vi.mock('../App.jsx', () => {
  return {
    default: () => <p>App page</p>,
  };
});

const entry = { initialEntries: ['/error'] };

it('Renders the error page', () => {
  const router = createMemoryRouter(routes, entry);

  render(<RouterProvider router={router} />);

  expect(screen.getByTestId('error-page')).toMatchSnapshot();
});

it("Renders the main page when 'here' link is clicked", async () => {
  const user = userEvent.setup();
  const router = createMemoryRouter(routes, entry);

  render(<RouterProvider router={router} />);

  const link = screen.getByRole('link', { name: 'here' });

  await user.click(link);

  expect(screen.getByText('App page')).toBeInTheDocument();
});
