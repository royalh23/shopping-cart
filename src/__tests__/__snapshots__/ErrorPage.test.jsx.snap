// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders the error page 1`] = `
<div
  class="_errorPage_175deb"
  data-testid="error-page"
>
  <h1>
    Oops, page not found
  </h1>
  <p>
    Click 
    <a
      href="/"
    >
      here
    </a>
     to get back to homepage.
  </p>
</div>
`;
